File: src/004.StackAndRecursion/071.SimplifyPath/71.简化路径.js
/*
 * @lc app=leetcode.cn id=71 lang=javascript
 *
 * [71] 简化路径
 */

// @lc code=start
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function (path) {
	const stack = [];
	path = path.split('/');

	for (const i of path) {
		if (i === '..' && stack.length) {
			stack.pop();
		} else if (i !== '.' && i !== '' && i !== '..') {
			stack.push(i);
		}
	}

	return '/' + stack.join('/');
};
// console.log(simplifyPath('/a/./b/../../c/'));
// @lc code=end

File: src/004.StackAndRecursion/071.SimplifyPath/71.简化路径.go
/*
 * @lc app=leetcode.cn id=71 lang=golang
 *
 * [71] 简化路径
 */

// @lc code=start
/*
func simplifyPath(path string) string {
	var stack []string
	pathArr := strings.Split(path, "/")

	for _, v := range pathArr {
		if v == ".." && len(stack) > 0 {
			stack = stack[:len(stack)-1]
		} else if v != "." && v != ".." && v != "" {
			stack = append(stack, v)
		}
	}

	return "/" + strings.Join(stack, "/")
}
*/
import (
	path2 "path"
)

func simplifyPath(path string) string {
	return path2.Clean(path)
}

// @lc code=end

